tests/resources/test_factorization.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for functions needed for two-electron tensor factorization.
+"""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+@pytest.mark.parametrize(
```

Comment:
Current tests seem good, but I wonder if we are being sufficiently comprehensive.

For example, I think we should check whether inputs have the right form/shape and test that we raise the appropriate warning

I think we should probably have tests for different error thresholds